geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/systems/WearableItemListener.kt

```diff
@@ -67,4 +72,18 @@ object WearableItemListener : Listener {
         currItem.subtract(1)
     }
 
+    @EventHandler
+    fun PlayerInteractAtEntityEvent.rightClickArmorStand() {
+        val item = player.inventory.itemInMainHand
+        val armorstand = rightClicked as ArmorStand
+
+        if (hand != EquipmentSlot.HAND) return
+        if (rightClicked.type != EntityType.ARMOR_STAND) return
+        item.toGearyFromUUIDOrNull()?.get<Hat>() ?: return
```

Comment:
Is this deliberate use of FromUUID? I'm pretty sure it doesn't handle instanced items

Reply:
Imean it worked when i tried so.
only other one is toGearyOrNull(player) but not really any difference between the two ingame

Reply:
Check the doc on each function, it should explain which one you should use by default

Reply:
Helmets arent player-instanced so iguess thats why it works.
I'll swap it to other one since might want player instanced hats in the future for some reason